src/Bicep.Core/Configuration/ConfigHelper.cs

```diff
@@ -1,29 +1,39 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+using Bicep.Core.FileSystem;
 using Microsoft.Extensions.Configuration;
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
-using System.Linq;
 using System.Reflection;
 
 namespace Bicep.Core.Configuration
 {
     public class ConfigHelper
     {
-        private const string SettingsFileName = "bicepconfig.json";
+        private const string bicepConfigResourceName = "Bicep.Core.Configuration.bicepconfig.json";
+        private readonly IFileResolver fileResolver;
 
         /// <summary>
         /// Property exposes the configuration root
         /// that is currently loaded
         /// </summary>
         public IConfigurationRoot Config { get; private set; }
```

Comment:
One question (non blocking, not related to this PR) - is there any reason why we are using the raw `IConfigurationRoot` interface? Would it be possible to user `ConfigurationBinder.Bind` to bind the `IConfigurationRoot` to an instance of a strongly typed config class?

Reply:
The linter rules currently read dynamically from the config: https://github.com/Azure/bicep/blob/de8b847579b19f5858f330b84d343849b7c520b4/src/Bicep.Core/Analyzers/Linter/Rules/NoHardcodedEnvironmentUrlsRule.cs#L40-L44

It would be nice to have a strongly-typed config class (e.g. for the alias work you're doing), but might need to keep some dynamic-ness for linters if we want 3rd parties to be able to author their own.